include/swift/AST/TypeOrExtensionDecl.h

```diff
@@ -0,0 +1,59 @@
+//===- TypeOrExtensionDecl.h - Swift Language Declaration ASTs -*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the TypeOrExtensionDecl struct, separately to Decl.h so
+// that this can be included in files that Decl.h includes.
```

Comment:
This probably doesn't actually work, because PointerUnion needs to know the alignment of the types in it to know where to stick spare bits.

Reply:
It seems to compile.

Reply:
It compiles as long as you also include Decl.h before the template actually has to be instantiated, I think.

Reply:
How do I address this then?

Reply:
Untangle the headers? Or…ah, it looks like we have swift/AST/TypeAlignments.h for this. ASTNode.h uses it in a similar way.

Reply:
Neat; changed it to use that header.